notebooks/labs37_notebooks/GenerateTweet_LSTM.ipynb

```diff
@@ -0,0 +1,1225 @@
+        "  text =[]\n",
+        "\n",
+        "  for _ in range(n_words):\n",
+        "    # text_to_sequences: provides a tuple of {words:integer}\n",
+        "    # we need only the integer number of seed_text and thus the 0th element\n",
+        "    encoded = tokenizer.texts_to_sequences([seed_text])[0]\n",
+        "    # encoded may have more than 50 words so we need to truncate it at 50th word and thus pad the sequence\n",
+        "    encoded = pad_sequences([encoded],maxlen = text_seq_length, truncating ='pre')\n",
+        "\n",
+        "    #predict class for each word which is an integer value\n",
+        "    y_predict = model.predict_classes(encoded)\n",
+        "\n",
+        "    predicted_word = ''\n",
+        "    # now we match the y_predict integer with the integer from the tokenizer\n",
+        "      # if we find a match we grab the related word as the predicted_word\n",
+        "    for word ,index in tokenizer.word_index.items():\n",
+        "      if index == y_predict:\n",
+        "        predicted_word = word\n",
+        "        break\n",
+        "\n",
+        "    # this predicted word is added to the seed text\n",
+        "    seed_text = seed_text + ' ' + predicted_word\n",
+        "    text.append(predicted_word)\n",
+        "  return \" \".join(text) "
+      ],
+      "execution_count": 26,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 106
+        },
+        "id": "ag8rQmGsutcn",
+        "outputId": "23d79d24-4f28-4e43-933d-098a59e0f3a7"
+      },
+      "source": [
+        "generate_text_seq(model,tokenizer,sequence_length,seed_text,30)"
+      ],
+      "execution_count": 27,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/engine/sequential.py:455: UserWarning: `model.predict_classes()` is deprecated and will be removed after 2021-01-01. Please use instead:* `np.argmax(model.predict(x), axis=-1)`,   if your model does multi-class classification   (e.g. if it uses a `softmax` last-layer activation).* `(model.predict(x) > 0.5).astype(\"int32\")`,   if your model does binary classification   (e.g. if it uses a `sigmoid` last-layer activation).\n",
+            "  warnings.warn('`model.predict_classes()` is deprecated and '\n"
+          ],
+          "name": "stderr"
+        },
+        {
+          "output_type": "execute_result",
+          "data": {
+            "application/vnd.google.colaboratory.intrinsic+json": {
+              "type": "string"
+            },
+            "text/plain": [
+              "'block of sw alder st one officer strikes a protester repeatedly with their baton that protester was later arrested at an antiice protest in the portland pearl district several police'"
```

Comment:
Did the model write this line?

Reply:
It did generate that line. Going by what I'm seeing as results, I wonder if we need perfect sentences to serve as input to the BERT model. If so then there is either more tweaking on this LSTM to be done or manually just grabbing snippets of the generated tweets.

Reply:
This looks better than I expected it would. I don't think we need "perfect sentences" for BERT; we need "admissible sentences," which is harder to figure out. This generated line is not perfect but one could interpret it as the user hastily first writing the location and moving on to write "one officer strikes a protester..." Tweets (and sentences contained) aren't always composed of perfect sentences, but they obviously need to make sense. What I wonder is whether we could generate hundreds of admissible sentences through this method, so having mass production - and whether (and if so at what point) it would cease to generate admissible sentences